src/string.cr

```diff
@@ -853,11 +853,13 @@ class String
     char_at(index) { yield }
   end
 
+  # Returns the `Char` at the given *index*, or raises `IndexError` if out of bounds.
```

Comment:
It would be good to add code examples to all of the methods, showing possible cases. In this particular method I'd show an example for `char_at` of an good index and a bad index.

Reply:
done